src/game/main.cpp

```diff
@@ -162,6 +171,203 @@ void Check_Windowed(int argc, char *argv[])
+            break;
+
+        case WM_KILLFOCUS:
+            if (g_theKeyboard != nullptr) {
+                g_theKeyboard->Reset_Keys();
+            }
+            if (g_theWin32Mouse != nullptr) {
+                g_theWin32Mouse->Set_Ignore_Events(true);
+            }
+            break;
+
+        case WM_PAINT:
+            if (g_gameNotFullscreen) {
+                PAINTSTRUCT ps{};
+                auto hDC = BeginPaint(hWnd, &ps);
+                if (g_splashImage != nullptr) {
+                    const auto old_dc = SaveDC(hDC);
+                    auto compat_dc = CreateCompatibleDC(hDC);
+                    auto ha = SelectObject(compat_dc, g_splashImage);
+                    BitBlt(hDC, 0, 0, 800, 600, compat_dc, 0, 0, SRCCOPY);
+                    SelectObject(compat_dc, ha);
+                    DeleteDC(compat_dc);
+                    RestoreDC(hDC, old_dc);
+                }
+                return 0; // vanilla returned 1
+            }
+            break;
+
+        case WM_CLOSE:
+            if (g_theGameEngine->Get_Quitting()) {
+                if (g_theKeyboard != nullptr) {
+                    g_theKeyboard->Reset_Keys();
+                }
+                if (g_theWin32Mouse != nullptr) {
+                    g_theWin32Mouse->Set_Ignore_Events(false);
+                }
+                break;
+            }
+            //[[fallthrough]]
+        case WM_QUERYENDSESSION:
+            g_theMessageStream->Append_Message(GameMessage::MessageType::MSG_META_DEMO_INSTANT_QUIT);
+            return 0;
+
+        case WM_ERASEBKGND:
+            if (!g_gameNotFullscreen) {
+                return 1;
+            }
+            break;
+
+        case WM_ACTIVATEAPP: {
+            bool activateState = wParam == TRUE;
+            if (activateState == g_gameActive) {
+                return 0;
+            }
+            g_gameActive = activateState;
+            if (g_theGameEngine != nullptr) {
+                g_theGameEngine->Set_Is_Active(activateState);
+            }
+            sub_73C740(activateState);
+            if (activateState && g_theWin32Mouse != nullptr) {
+                g_theWin32Mouse->Set_Cursor(g_theWin32Mouse->Get_Mouse_Cursor());
+            }
+            return 0;
+        }
+        case WM_NCHITTEST:
+            if (!g_gameIsWindowed) {
+                return HTCLIENT;
+            }
+            break;
+
+        case WM_KEYDOWN:
+            if (wParam == VK_ESCAPE) {
+                PostQuitMessage(EXIT_SUCCESS);
+            }
+            return 0;
+
+        case WM_SYSCOMMAND:
+            switch (wParam) {
+                case SC_MAXIMIZE:
+                case SC_SIZE:
+                case SC_MOVE:
+                case SC_KEYMENU:
+                case SC_MONITORPOWER:
+                    if (!g_gameIsWindowed) {
+                        return 0; // vanilla returned 1
+                    }
+            }
+            break;
+
+        case WM_MOUSEMOVE:
```

Comment:
seems like if you pass Add_Win32_Event msg instead of WM_MOUSEMOVE its handled the same as mouse wheel?

Reply:
yes do you want me to combine them?

Reply:
yeah

Reply:
Ah forgot one is using clientrect the other is using windowrect so they are subtly different.